src/Formatter/ClassNameFormatter.php

```diff
@@ -28,15 +28,26 @@ public function commandHandled($command)
 
     /**
      * @param object $command
-     * @param Exception $e
      * @return string|null
      */
-    public function commandFailed($command, Exception $e)
+    public function commandFailed($command)
     {
-        $commandClass = get_class($command);
-        $exceptionClass = get_class($e);
-        $exceptionMessage = $e->getMessage();
+        return 'Command failed: ' . get_class($command);
+    }
 
-        return "Command failed: {$commandClass} threw the exception {$exceptionClass} ({$exceptionMessage})";
+    /**
+     * {@inheritDoc}
+     */
+    public function commandContext($command)
+    {
+        return ['class' => get_class($command)];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function completeCommandContextOnFailure(array $currentContext, \Exception $e)
+    {
+        return ['error' => ['class' => get_class($e), 'message' => $e->getMessage()]] + $currentContext;
```

Comment:
Hmm, maybe we could split this into two dedicated methods, so this one only returns the error stuff? commandContext for just the command stuff and failureContext (or exceptionContext) for the exception.

I know there's some folks who only want to log the exception class+message, not the entire command data blob. Splitting that gives them the option to write their own formatter. We can just array_merge it in the formatters we use (maybe with the top level keys set there to avoid accidental merge collisions).

Also, it might be better not to typehint this since we're still supporting 5.6 and we want to make sure we allow Throwables in 7. We can typehint for this in the docblock so we at least have soft enforcement.